include/bout/mesh.hxx

```diff
@@ -960,6 +1000,10 @@ protected:
   /// Pointer to the global MPI wrapper, for convenience
   MpiWrapper* mpi = nullptr;
 
+public:
+  // Switch for communication of corner guard and boundary cells
+  const bool include_corner_cells;
```

Comment:
I don't immediately see how to move the logic into `BoutMesh::send` - the `include_corner_cells` case needs to wait twice (in the current implementation, this may change if/when I revert back to the version with simultaneous communication of x-, y- and corner-guard cells).

On the other hand, I expect `include_corner_cells` might be wanted, at least for checks, in other places eventually. For example in `D2DXDY` we could have a simpler, possibly more consistent, implementation when we have consistent corner cells; we might even want to only allow `D2DXDY` to be called when `include_corner_cells == true`.